src/platform_impl/macos/view.rs

```diff
@@ -315,10 +318,25 @@ declare_class!(
 
             // Empty string basically means that there's no preedit, so indicate that by sending
             // `None` cursor range.
+            let mut lowerbound: usize = preedit_string.len();
+            let mut upperbound: usize = lowerbound;
+            if !preedit_string.is_empty() {
+                unsafe {
+                    let new_bounds = get_actual_range_bounds(&preedit_string, selected_range);
+                    match new_bounds {
+                        None => {},
+                        Some(fetched_value) => {
+                            lowerbound = fetched_value.0;
+                            upperbound = fetched_value.1;
+                        }
+                    }
+                }
```

Comment:
I think this statement can be simplified with `if-let` syntax, as the `None` branch is empty.

```suggestion
                if let Some(fetched_value) = new_bounds {
	            lowerbound = fetched_value.0;
	            upperbound = fetched_value.1;
                }
```